specs/decimal/decimal.md

```diff
@@ -197,66 +339,307 @@ Windows App SDK provides a native language decimal data type for C++ as the
+    char to_char() const;
+    std::int16_t to_int16() const;
+    std::int32_t to_int32() const;
+    std::int64_t to_int64() const;
+    std::uint8_t to_uint8() const;
+    std::uint16_t to_uint16() const;
+    std::uint32_t to_uint32() const;
+    std::uint64_t to_uint64() const;
+    float to_float() const;
+    double to_double() const;
+    long to_long() const;
+    unsigned long to_ulong() const;
+    std::wstring to_string() const;
+    std::wstring to_string(const LCID locale) const;
+#if defined(__WINSTRING_H_)
+    HSTRING to_HSTRING() const;
+    HSTRING to_HSTRING(const LCID locale) const;
+#endif // defined(WINRT_BASE_H)
+#if defined(WINRT_BASE_H)
+    winrt::hstring to_hstring() const;
+    winrt::hstring to_hstring(const LCID locale) const;
+#endif // defined(WINRT_BASE_H)
+
+    bool operator==(const decimal& value) const;
+    bool operator!=(const decimal& value) const;
+    bool operator< (const decimal& value) const;
+    bool operator<=(const decimal& value) const;
+    bool operator> (const decimal& value) const;
+    bool operator>=(const decimal& value) const;
+    int compare(const decimal& value) const;
+
+    bool operator==(const DECIMAL& value) const;
+    bool operator!=(const DECIMAL& value) const;
+    bool operator< (const DECIMAL& value) const;
+    bool operator<=(const DECIMAL& value) const;
+    bool operator> (const DECIMAL& value) const;
+    bool operator>=(const DECIMAL& value) const;
+    int compare(const DECIMAL& value) const;
+
+    /// Return true if this is valid.
+    bool is_valid();
+
+    /// Return true if value is valid.
+    static constexpr bool is_valid(const DECIMAL& value);
+
+    /// Return the scaling factor of the value (the number of decimal digits).
+    /// @return the scaling factor, ranging from 0 to max_scale().
+    std::uint32_t scale() const;
+
+    /// Return the sign of the value.
+    /// @return 0 if this os zero, <0 if this is less than zero or >0 if this is greater than zero.
+    std::int32_t sign() const;
+
+    /// Return the maximum scaling factor
+    static constexpr std::uint32_t max_scale();
+
+    /// Return the maximum value (79,228,162,514,264,337,593,543,950,335).
+    static constexpr decimal max_value();
+
+    /// Return the minimum value (-79,228,162,514,264,337,593,543,950,335).
+    static constexpr decimal min_value();
+
+    decimal operator+() const;
+
+    decimal operator-() const;
+
+    decimal abs() const;
+
+    /// Return the integral digits; any fractional digits are discarded.
+    decimal truncate() const;
+
+    /// Return the integral digits rounded down to -infinity; any fractional digits are discarded.
+    decimal floor() const;
+
+    /// Return the integral digits rounded up to +infinity; any fractional digits are discarded.
+    decimal ceil() const;
+
+    /// Return this clamped to the inclusive range of min and max.
+    /// @return this if min <= this <= max, or min if this < min, or max if max < this.
```

Comment:
Three more cases not covered: What happens if `min > max`?

Reply:
DISCUSSION: `min > max` throws an exception

CONSIDER: Review `std::clamp`

Reply:
CONSIDER: Add doccomment `@throw...` for the methods that can throw exceptions (only a handful)